meta-lmp-base/classes/lmp.bbclass

```diff
@@ -140,6 +140,36 @@ IMAGE_CMD:ostree:append () {
        fi
 }
 
+# Post ext4 generation logic for luks2 based disk encryption
+IMAGE_CMD:ota-ext4:append () {
+	if [ "${OSTREE_OTA_EXT4_LUKS}" = "1" ]; then
+		if [ -z "${OSTREE_OTA_EXT4_LUKS_PASSPHRASE}" ]; then
+			bbfatal "Unable to find passphrase for LUKS-based ota-ext4 (define OSTREE_OTA_EXT4_LUKS_PASSPHRASE)"
+		fi
+		cp ${IMGDEPLOYDIR}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.ota-ext4 ${IMGDEPLOYDIR}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.ota-ext4-orig
+
+		block_size=`dumpe2fs -h ${IMGDEPLOYDIR}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.ota-ext4 | grep "^Block size" | cut -d ':' -f 2 | tr -d ' '`
+		block_count=`dumpe2fs -h ${IMGDEPLOYDIR}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.ota-ext4 | grep "^Block count" | cut -d ':' -f 2 | tr -d ' '`
+		luks_size=33554432 # 32m
```

Comment:
My only comment here is, do you expect to hard code this to 32 megabytes. and is that only for the initial image and will expand correct on first boot?

Reply:
Yes, this is the recommended amount for the luks2 metadata. When doing the resize on first boot it takes this into account as well (which is why we need to call cryptsetup resize as well).